rotkehlchen/chain/ethereum/modules/eigenlayer/decoder.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any
+
+from rotkehlchen.chain.ethereum.modules.eigenlayer.constants import (
+    CPT_EIGENLAYER,
+    DEPOSIT_TOPIC,
+    EIGENLAYER_CPT_DETAILS,
+    EIGENLAYER_STRATEGY_MANAGER,
+)
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.resolver import ethaddress_to_identifier
+from rotkehlchen.history.events.structures.evm_event import EvmProduct
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address
+
+
+class EigenlayerDecoder(DecoderInterface):
+
+    def _decode_deposit(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != DEPOSIT_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        depositor = hex_or_bytes_to_address(context.tx_log.data[0:32])
+        token_addr = hex_or_bytes_to_address(context.tx_log.data[32:64])
+        token_identifier = ethaddress_to_identifier(address=token_addr)
+        strategy = hex_or_bytes_to_address(context.tx_log.data[64:96])
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.asset == token_identifier and
+                event.location_label == depositor
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                asset = event.asset.resolve_to_crypto_asset()
+                event.notes = f'Deposit {event.balance.amount} {asset.symbol} in EigenLayer'
```

Comment:
Question. I am not very much into eigenlayer so no idea how it works. Bit does the asset symbol give enough info on what you are depositing to? Or should the address also be in the notes? If it does not need to it's better as it's not that noisy.

Reply:
Yes with the symbol is enough for what I know. You deposit any LST and you start receiving points in their system